lcls_tools/superconducting/sc_ssa.py

```diff
@@ -0,0 +1,342 @@
+            # number of times before raising an error)
+            self.reset()
+
+            print(f"Turning {self} on")
+            self.turn_on_pv_obj.put(1)
+
+            while not self.is_on:
+                self.cavity.check_abort()
+                print(f"waiting for {self} to turn on")
+                sleep(1)
+
+        if self.cavity.cryomodule.is_harmonic_linearizer:
+            self.ps_volt_setpoint2_pv_obj.put(utils.HL_SSA_PS_SETPOINT)
+            self.ps_volt_setpoint1_pv_obj.put(utils.HL_SSA_PS_SETPOINT)
+
+        print(f"{self} on")
+
+    @property
+    def turn_off_pv_obj(self) -> PV:
+        if not self._turn_off_pv_obj:
+            self._turn_off_pv_obj = PV(self.turn_off_pv)
+        return self._turn_off_pv_obj
+
+    def turn_off(self):
+        if self.is_on:
+            print(f"Turning {self} off")
+            self.turn_off_pv_obj.put(1)
+
+            while self.is_on:
+                self.cavity.check_abort()
+                print(f"waiting for {self} to turn off")
+                sleep(1)
+
+        print(f"{self} off")
+
+    @property
+    def reset_pv_obj(self) -> PV:
+        if not self._reset_pv_obj:
+            self._reset_pv_obj = PV(self.reset_pv)
+        return self._reset_pv_obj
+
+    def reset(self):
+        reset_attempt = 0
+        while self.is_faulted:
+            print(f"Resetting {self}...")
+            self.reset_pv_obj.put(1)
+
+            while self.is_resetting:
+                sleep(1)
+
+            if self.is_faulted and reset_attempt >= utils.INTERLOCK_RESET_ATTEMPTS:
+                raise utils.SSAFaultError(
+                    f"{self} failed to reset {utils.INTERLOCK_RESET_ATTEMPTS}x"
+                )
+
+            reset_attempt += 1
+
+        print(f"{self} reset")
+
+    def start_calibration(self):
+        if not self._calibration_start_pv_obj:
+            self._calibration_start_pv_obj = PV(self.calibration_start_pv)
+        self._calibration_start_pv_obj.put(1)
+
+    @property
+    def calibration_status(self):
+        if not self._calibration_status_pv_obj:
+            self._calibration_status_pv_obj = PV(self.calibration_status_pv)
+        return self._calibration_status_pv_obj.get()
+
+    @property
+    def calibration_running(self) -> bool:
+        return self.calibration_status == utils.SSA_CALIBRATION_RUNNING_VALUE
+
+    @property
+    def calibration_crashed(self) -> bool:
+        return self.calibration_status == utils.SSA_CALIBRATION_CRASHED_VALUE
+
+    @property
+    def cal_result_status_pv_obj(self) -> PV:
+        if not self._cal_result_status_pv_obj:
+            self._cal_result_status_pv_obj = PV(self.cal_result_status_pv)
+        return self._cal_result_status_pv_obj
+
+    @property
+    def calibration_result_good(self) -> bool:
+        return self.cal_result_status_pv_obj.get() == utils.SSA_RESULT_GOOD_STATUS_VALUE
+
+    def run_calibration(self, save_slope: bool = False):
+        """
```

Comment:
I'm trying to think about whether we want to run a calibration any time we turn on an SSA and I think you might be right, because if a cavity/SSA hasn't been used in a while, we probably wanna run auto setup anyway 🤔

Reply:
Is this getting moved out?